scripts/generate_artifacts.js

```diff
@@ -0,0 +1,71 @@
+/**
+* Takes files in metadata/*.json and generates final metadata file by adding:
+*  - Build results (ABI and Bytecode)
+*  - Migration results (deployed instance address)
+*  - Natspec and auth role per function
+*/
+
+const fs = require('fs')
+const path = require('path')
+const { signatures } = require('../test/helpers/web3')
+const inspector = require('solidity-inspector')
+
+const getContract = x => artifacts.require(x)
+const flatten = x => [].concat.apply([], x)
+const contractName = p => path.basename(p).split('.')[0]
+
+const metadataDir = './metadata'
+const artifactsDir = './artifacts'
+
+const analizeContract = (f) => {
+    const filePath = path.join(process.cwd(), 'contracts', f)
+    const functions = inspector.parseFile(filePath).toJSON().functions
+
+    return Object.keys(functions)
+        .map(k => functions[k])
+        .filter(f => f.accessModifier != 'internal' && f.accessModifier != 'private')
+}
+
+const processFile = name => {
```

Comment:
Can we write a comment or two about what processing a file in this context means, or give the function a more meaningful name?